crates/relayer/src/config/filter.rs

```diff
@@ -24,21 +59,70 @@ pub enum PacketFilter {
     AllowAll,
 }
 
-impl Default for PacketFilter {
+// Currently only filtering on `recv_fee` is authorized
+#[derive(Clone, Debug, Default, PartialEq, Serialize, Deserialize)]
+pub struct FeesFilters {
+    recv: Vec<FeeInformation>,
+}
+
+impl FeesFilters {
+    pub fn new(recv: Vec<FeeInformation>) -> Self {
+        Self { recv }
+    }
+
+    pub fn should_relay(&self, event_type: IbcEventType, amounts: Vec<Coin<String>>) -> bool {
+        match event_type {
+            IbcEventType::SendPacket => {
+                for coin in amounts {
+                    if self
+                        .recv
+                        .iter()
+                        .any(|e| e.check_valid_fee(coin.amount.0, coin.denom.clone()))
+                    {
+                        return true;
+                    }
+                }
+                false
+            }
+            _ => true,
+        }
+    }
+}
+
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
+pub struct FeeInformation {
+    amount: u64,
+    denom: Option<String>,
+}
+
+impl FeeInformation {
+    pub fn new(amount: u64, denom: Option<String>) -> Self {
+        Self { amount, denom }
+    }
+
+    pub fn check_valid_fee(&self, other_amount: U256, other_denom: String) -> bool {
+        match self.denom.clone() {
+            Some(denom) => U256::from(self.amount) <= other_amount && denom.eq(&other_denom),
+            None => U256::from(self.amount) <= other_amount,
```

Comment:
Let's flip the comparison (using >= instead of <=) to improve readability a little bit